src/trezor/ui/num_pad.py

```diff
@@ -0,0 +1,48 @@
+from trezor import ui
+from trezor.ui import display
+from trezor.ui.button import BTN_CLICKED, Button
+
+ITEMS_PER_PAGE = 11
+
+
+def digit_area(i):
+    return ui.grid(i + 3)  # skip the first line
+
+
+class NumPad(ui.Widget):
+    def __init__(self, label: str, start: int, end: int):
+        """
+        Generates a numpad with numbers from `start` to `end` excluding.
+        """
+        display.clear()
+        self.label = label
+        self.start = start
+        self.end = end
+        self.page = 0
+
+        self._generate_buttons()
+
+    def render(self):
+        for btn in self.buttons:
+            btn.render()
+
+        # header label
+        display.text_center(ui.WIDTH // 2, 36, self.label, ui.BOLD, ui.GREY, ui.BG)
+
+    def touch(self, event, pos):
+        for btn in self.buttons:
+            if btn.touch(event, pos) == BTN_CLICKED:
+                if "+" in btn.content:
+                    self.page += 1
+                    self._generate_buttons()
+                else:
+                    return btn.content
+
+    def _generate_buttons(self):
+        display.clear()
```

Comment:
Please add a code comment explaining why we are clearing the screen here.

Reply:
Hm, because we need to redraw the buttons with other values. I can comment, but it makes sense, doesn't it?

Reply:
This is what I get if I don't:

![emu00000001](https://user-images.githubusercontent.com/1835345/54211082-f973f500-44e0-11e9-91a4-dd85dbbfa62f.png)

The last row should not be there at all.